clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ConsumptionTask.java

```diff
@@ -50,6 +52,16 @@ class ConsumptionTask implements Runnable {
   private final AggKafkaConsumerServiceStats aggStats;
   private final ConsumerSubscriptionCleaner cleaner;
 
+  /**
+   * Maintain rate counter with default window size to calculate the message and bytes rate at topic partition level.
+   */
+  private final Map<PubSubTopicPartition, Rate> messageRatePerTopicPartition = new VeniceConcurrentHashMap<>();
```

Comment:
In this [PR](https://github.com/linkedin/venice/pull/832/files), we've added store-level metrics for these already. Do we need to obtain them again at the topic-partition level?

Reply:
I want to deduplicate metrics since we are already emitting too much metrics.

Reply:
oh, just took another look, so your codes are not emitting any metrics but still use sensors to collect them.

Reply:
The metric you added will be emitted by all consumer thread to one store metric, the end result shown on ingraph is the store level metric. For previous siteup issue, we find a lot of cases only 1 or few topic partitions per store is slow and its slowness is caused by sharing with other store's partition (noisy neighboor). For my implementation here, I just use the Rate() to do the window calculation for each topic partition, this information will not be emitted to ingraph. We just kept them in server host, with admin tool we can obtain them for debugging.

Reply:
Can you plz add some comments saying these data are for not monitoring purpose?

Reply:
Sure, I can add that.